accepted/00000-salt-ssh-with-salt-bundle.md

```diff
@@ -0,0 +1,77 @@
+
+The "Pre flight script" calculates the bootstrap repository URL in a similar manner to the existing boostrap.sh script, based on the operating system and OS architecture of the system it is executed on.
+It requires one parameter from the Uyuni server: (host, port) (entry point) for SSH connections through tunnels.
+The salt codebase need to be modified to pass additional parameters, otherwise we have to generate pre flight script individually for each managed system.
+
+## Deploying Salt Bundle to the managed system
+
+We don't need to install Salt Bundle package on the managed system.
+We need to extract the virtual environment of the bundle from the package.
+Pre flight script is deploying it as described above.
+
+## Using proper salt codebase
+
+On deploying Salt Bundle with pre flight script, without modification salt-thin (`salt-thin` is a tarball with salt codebase created by the server to deploy salt on salt-ssh managed system: https://docs.saltproject.io/en/latest/ref/runners/all/salt.runners.thin.html) deployment, we will get two different codebases of salt deployed on the managed system (one codebase is from salt bundle package for the OS+arch and the other one is from salt-master delivered with salt-thin tarball).
+As the sources of these two codebases are different (update channel of the Server and client tools relevant for the system) these codebases are not in sync.
+We can prevent deploying salt-thin in case if Salt Bundle has been deployed already and use it as preferred salt codebase to handle salt-ssh session with on managed system side.
+In most cases the client tools channels are updating more frequently than the server and for the minions (not salt-ssh systems) we are using the salt codebase from client tools, it's better to use Salt Bundle codebase for salt-ssh systems.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+  * Salt Bundle consumes more space than salt-thin on managed system
+  * Salt is creating additional SSH connection to run the pre flight script 
+  * Pre flight script requires an access to the bootstrap repository (salt-thin is deployed with ssh, no http(s) connection needed for salt-ssh)
+  * We need to modify salt codebase to handle Salt Bundle the proper way
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Prepare Salt Bundle for each OS and architecture on the Server, but it just make pre flight script bit smaller, the overall complexity is higher due to the bunch of additional changes on server side.
+- With not doing this the systems having not having sufficient Python version can't be managed with Salt-SSH and bootstrapped with web UI
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- The way to update Salt Bundle used for Salt-SSH on the managed system and validate it (cases when we need to redeploy it).
+  It can be easily solved with including sha256 to `venv-enabled-ARCH.txt` in the root of bootstrap repo.
```

Comment:
It looks like this unresolved question is now resolved, or at least we have a tentative answer. Please move this information in the RFC body, leaving only really-still-open questions here.

Reply:
Yup, I realized it on discussing with @SchoolGuy and PR appending the hash to `venv-enabled-ARCH.txt` was already merged. Will move it.

Reply:
But the validation of the code deployed (is it complete and unchanged) is still unknown.